opensrp-app/src/main/java/org/smartregister/repository/P2PSenderTransferDao.java

```diff
@@ -53,10 +53,10 @@ public JsonData getJsonData(@NonNull DataType dataType, long l, int i) {
         } else if (dataType.getName().equals(task.getName())) {
             return CoreLibrary.getInstance().context()
                     .getTaskRepository().getTasks(l, i);
-        } else if (dataType.getName().equals(foreignClient.getName())) {
+        } else if (dataType.getName().equals(foreignClient.getName()) && CoreLibrary.getInstance().context().hasForeignEvents()) {
```

Comment:
refactor so that `CoreLibrary.getInstance().context().hasForeignEvents()` is at the start